app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -1,43 +1,85 @@
 package org.oppia.app.parser
 
 import org.oppia.app.model.Fraction
+import org.oppia.app.topic.FractionParsingErrors
 import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Integer.parseInt
+import java.util.regex.Pattern
 
 /** This class contains method that helps to parse string to fraction. */
 class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d)+$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d)+ ?(\d+) ?/ ?(\d)+$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
+  private val fractionRegex = """^\s*-?\s*((\d*\s*\d+\s*\/\s*\d+)|\d+)\s*$""".toRegex()
```

Comment:
Instead of introducing a separate regex, let's take parseFraction's contents and move it to a helper that returns a null Fraction if the parsing fails. Then, we can rely on the same parsing logic in the error checking method below and don't have to worry about multiple regex patterns getting out of date.

Reply:
Done